src/main/java/bio/terra/cli/command/Status.java

```diff
@@ -1,29 +1,64 @@
 package bio.terra.cli.command;
 
-import bio.terra.cli.context.GlobalContext;
-import bio.terra.cli.context.WorkspaceContext;
-import java.util.concurrent.Callable;
+import bio.terra.cli.command.helperclasses.CommandSetup;
+import bio.terra.cli.command.helperclasses.FormatFlag;
+import bio.terra.cli.context.ServerSpecification;
+import bio.terra.workspace.model.WorkspaceDescription;
+import picocli.CommandLine;
 import picocli.CommandLine.Command;
 
 /** This class corresponds to the second-level "terra status" command. */
 @Command(name = "status", description = "Print details about the current workspace.")
-public class Status implements Callable<Integer> {
+public class Status extends CommandSetup {
 
+  @CommandLine.Mixin FormatFlag formatFlag;
+
+  /** Build the return value from the global and workspace context. */
   @Override
-  public Integer call() {
-    GlobalContext globalContext = GlobalContext.readFromFile();
-    WorkspaceContext workspaceContext = WorkspaceContext.readFromFile();
+  protected void execute() {
+    StatusReturnValue statusReturnValue =
+        new StatusReturnValue(globalContext.server, workspaceContext.terraWorkspaceModel);
+    formatFlag.printReturnValue(statusReturnValue, returnValue -> this.printText(returnValue));
+  }
+
+  /** POJO class for printing out this command's output. */
+  public static class StatusReturnValue {
+    // global server context = service uris, environment name
+    public final ServerSpecification server;
 
-    System.out.println("Terra server: " + globalContext.server.name);
+    // workspace description object returned by WSM
+    public final WorkspaceDescription workspace;
+
+    public StatusReturnValue(ServerSpecification server, WorkspaceDescription workspace) {
+      this.server = server;
+      this.workspace = workspace;
+    }
+  }
+
+  /**
+   * Print this command's output in text format.
+   *
+   * @param returnValue command return value object
```

Comment:
optional nit: I don't think this javadoc adds much on this or other commands, but up to you.

Reply:
I prefer to have at least a brief description on all methods. Since these are `private` methods, I'm good with removing the `@param` though.